Added textstyle to selectionTitle

JanLouisP opened the pull request:
Added TextStyle for SelectionTitle. (Needed for Darktheme)

JanLouisP committed:
created a map literal for accessing key of properties in CustomPickerHandler

JanLouisP committed:
reNamed propertyTitle -> propertyKey

JanLouisP committed:
exposed SchemaProperty instead of Map

JanLouisP committed:
Added textstyle to selectionTitle

Ayman-Barghout reviewed (commented):
Would it be better to just set text style from text theme and override that on the dark theme text theme maybe?

JanLouisP commented:
> Would it be better to just set text style from text theme and override that on the dark theme text theme maybe?

I think I dont understand what you mean. I did it like it was done with all the other properties, title, description etc. We override it in done_flutter with for example: description: `Theme.of(context).textTheme.titleMedium!.apply(color: themeColor.uiPrimary),`

JanLouisP committed:
Switched to Theme.of(context)

JanLouisP committed:
removed comment

applied Textcolor

applied Textcolor

ErrorColor

I applied Colors.red, since I don't know, can we access the extentionsColor uiNegative?

applied Textcolor

changed Errorcolor

applied Textcolor

applied Textcolor

applied Errorcolor

applied Textcolor

applied Textcolor

applied Errorcolor

applied Textcolor

applied Errortextstyle

Ayman-Barghout commented on lib/src/builder/field_header_widget.dart:
```diff
@@ -13,13 +12,13 @@ class FieldHeader extends StatelessWidget {
           '${property.title} ${property.required ? "*" : ""}',
-          style: WidgetBuilderInherited.of(context).uiConfig.fieldTitle,
+          style: Theme.of(context).textTheme.bodySmall,
         ),
         if (description != null && description.isNotEmpty) ...[
           const SizedBox(height: 4),
           Text(
             description,
-            style: WidgetBuilderInherited.of(context).uiConfig.fieldDescription,
+            style: Theme.of(context).textTheme.bodyMedium,
```
It's kinda weird that the title has `bodySmall` while the description (smaller text, or less bold) has `bodyMedium`, maybe something we didn't consider when we were updating our text themes?

JanLouisP replied:
jop, I need to switch them. I propably do that in the my Darktheme implementation branch and when I'm done I do switch it in here.
<img width="182" alt="image" src="https://github.com/doneservices/flutter_jsonschema_builder/assets/124917097/01be4c60-d374-49b7-96ce-9efcd44b00f7">

Ayman-Barghout replied:
Yeah, we can also change it here as this won't directly affect the app unless we fetch these changes, then update it in our theming PR, so we can merge this one :)

JanLouisP replied:
I switched them bodyMedium in https://github.com/doneservices/done-flutter/pull/1589 and also here now.

JanLouisP replied:
Could we also Cherrypick this commit from https://github.com/doneservices/done-flutter/pull/1589 into a new branch and merge it in main?

Ayman-Barghout commented:
Great work and looks good to me except for that comment, let me know when it's ready for re-review or if you think otherwise 🙏

JanLouisP committed:
switched bodySmall <-> bodyMedium

JanLouisP committed:
Merge branch 'next' into jan/textstyle-for-select-title

Ayman-Barghout commented on lib/src/fields/text_form_field.dart:
```diff
@@ -90,14 +90,14 @@ class _TextJFormFieldState extends State<TextJFormField> {
             },
             style: widget.property.readOnly
                 ? const TextStyle(color: Colors.grey)
-                : WidgetBuilderInherited.of(context).uiConfig.label,
+                : Theme.of(context).textTheme.titleMedium,
```
I think it would be better here to use `apply` or `copyWith` on the `titleMedium` theme text style and change its color for disabled, otherwise titles will have different text style when read only which may look weird.

Ayman-Barghout commented on lib/src/fields/radio_button_form_field.dart:
```diff
@@ -101,9 +100,7 @@ class _RadioButtonJFormFieldState extends State<RadioButtonJFormField> {
-                                : WidgetBuilderInherited.of(context)
-                                    .uiConfig
-                                    .label),
+                                : Theme.of(context).textTheme.titleMedium),
```
Same as the above comment

Ayman-Barghout commented on lib/src/fields/text_form_field.dart:
```diff
@@ -90,14 +90,14 @@ class _TextJFormFieldState extends State<TextJFormField> {
                   : null,
               labelStyle: const TextStyle(color: Colors.blue),
-              errorStyle: WidgetBuilderInherited.of(context).uiConfig.error,
+              errorStyle: Theme.of(context).textTheme.bodyMedium,
```
This is an error style, so maybe we should apply a negative or error color to this?

Ayman-Barghout commented on lib/src/fields/number_form_field.dart:
```diff
@@ -67,7 +67,7 @@ class _NumberJFormFieldState extends State<NumberJFormField> {
           },
           style: widget.property.readOnly
               ? const TextStyle(color: Colors.grey)
-              : WidgetBuilderInherited.of(context).uiConfig.label,
+              : Theme.of(context).textTheme.titleMedium,
```
Same as the first comment

Ayman-Barghout commented on lib/src/fields/number_form_field.dart:
```diff
@@ -88,7 +88,7 @@ class _NumberJFormFieldState extends State<NumberJFormField> {
                     ? widget.property.help
                     : null,
-            errorStyle: WidgetBuilderInherited.of(context).uiConfig.error,
+            errorStyle: Theme.of(context).textTheme.bodyMedium,
```
This is also error style, maybe some error color from theme?

Ayman-Barghout commented on lib/src/fields/file_form_field.dart:
```diff
@@ -96,7 +96,7 @@ class _FileJFormFieldState extends State<FileJFormField> {
                         overflow: TextOverflow.ellipsis,
                         style: widget.property.readOnly
                             ? const TextStyle(color: Colors.grey)
-                            : widgetBuilderInherited.uiConfig.label),
+                            : Theme.of(context).textTheme.titleMedium),
```
Same as the first comment

Ayman-Barghout commented on lib/src/fields/dropdown_oneof_form_field.dart:
```diff
@@ -150,7 +152,7 @@ class _SelectedFormFieldState extends State<DropdownOneOfJFormField> {
                 item.title ?? '',
                 style: widget.property.readOnly
                     ? const TextStyle(color: Colors.grey)
-                    : WidgetBuilderInherited.of(context).uiConfig.label,
+                    : Theme.of(context).textTheme.titleMedium,
```
Same as the first comment

Ayman-Barghout commented on lib/src/fields/dropdown_oneof_form_field.dart:
```diff
@@ -110,8 +113,7 @@ class _SelectedFormFieldState extends State<DropdownOneOfJFormField> {
               onSaved: widget.onSaved,
               decoration: InputDecoration(
-                  errorStyle:
-                      WidgetBuilderInherited.of(context).uiConfig.error),
+                  errorStyle: Theme.of(context).textTheme.bodyMedium),
```
This is also an error style, maybe use a negative color on the text style.

Ayman-Barghout commented on lib/src/fields/checkbox_form_field.dart:
```diff
@@ -60,7 +59,7 @@ class _CheckboxJFormFieldState extends State<CheckboxJFormField> {
                     widget.property.title,
                     style: widget.property.readOnly
                         ? const TextStyle(color: Colors.grey)
-                        : WidgetBuilderInherited.of(context).uiConfig.label,
+                        : Theme.of(context).textTheme.titleMedium,
```
Same as first comment.

Ayman-Barghout commented on lib/src/fields/date_form_field.dart:
```diff
@@ -67,7 +67,7 @@ class _DateJFormFieldState extends State<DateJFormField> {
           readOnly: widget.property.readOnly,
           style: widget.property.readOnly
               ? const TextStyle(color: Colors.grey)
-              : WidgetBuilderInherited.of(context).uiConfig.label,
+              : Theme.of(context).textTheme.titleMedium,
```
Same as first comment.

Ayman-Barghout commented on lib/src/fields/date_form_field.dart:
```diff
@@ -92,7 +92,7 @@ class _DateJFormFieldState extends State<DateJFormField> {
               icon: const Icon(Icons.date_range_outlined),
               onPressed: widget.property.readOnly ? null : _openCalendar,
             ),
-            errorStyle: WidgetBuilderInherited.of(context).uiConfig.error,
+            errorStyle: Theme.of(context).textTheme.bodyMedium,
```
Error style

Ayman-Barghout commented on lib/src/fields/dropdown_form_field.dart:
```diff
@@ -87,9 +90,9 @@ class _DropDownJFormFieldState extends State<DropDownJFormField> {
               onSaved: widget.onSaved,
               style: widget.property.readOnly
                   ? const TextStyle(color: Colors.grey)
-                  : WidgetBuilderInherited.of(context).uiConfig.label,
+                  : Theme.of(context).textTheme.titleMedium,
```
Same as first comment

Ayman-Barghout commented on lib/src/fields/dropdown_form_field.dart:
```diff
@@ -87,9 +90,9 @@ class _DropDownJFormFieldState extends State<DropDownJFormField> {
+                  : Theme.of(context).textTheme.titleMedium,
               decoration: InputDecoration(
-                errorStyle: WidgetBuilderInherited.of(context).uiConfig.error,
+                errorStyle: Theme.of(context).textTheme.bodyMedium,
```
Error style

Ayman-Barghout reviewed (commented):
Great work on this one @JanLouisP, sorry for lots of comments, it's a small issue that needs to be fixed and it was an oversight on my part where I didn't account for the read-only text style and just left what was there.
Also, a small note regarding the error style's color, please ask for re-review once it's fixed and this PR is ready to go.

JanLouisP committed:
Merge branch 'jan/textstyle-for-select-title' of https://github.com/doneservices/flutter_jsonschema_builder into jan/textstyle-for-select-title

JanLouisP commented:
I did squash into parent. Thats why the commit looks like that 😬

Ayman-Barghout commented on lib/src/fields/date_form_field.dart:
```diff
@@ -92,7 +95,10 @@ class _DateJFormFieldState extends State<DateJFormField> {
+            errorStyle: Theme.of(context)
+                .textTheme
+                .bodyMedium!
+                .apply(color: Colors.red),
```
Maybe we should depend on the theme's error color instead of hardcoded red?

JanLouisP committed:
changed Hardcoded Color to Theme

Ayman-Barghout reviewed (approved):
Great work 🎉 ! It'd be great if we can check that everything looks good in the app after all these changes 🙂
